source/iopipe/json/serialize.d

```diff
@@ -65,12 +77,20 @@ private void onField(P, JT, T, size_t N)(ref P policy, ref JT tokenizer, ref T i
     static assert(N == SerializableMembers!T.length);
     alias members = SerializableMembers!T;
     alias ignoredMembers = AllIgnoredMembers!T;
+
     // Check each member to see if it matches
     switch(key)
     {
         static foreach(i, memberName; members) {
             { // Add a block scope to contain each declaration, avoiding duplicate jsonName
               // declarations in the foreach loop.
+              static if(hasUDA!(__traits(getMember, T, memberName), extras))
+              {
+                  // this is the extras member, it holds any extra data that was not
+                  // specified as a member.
+                  static assert(is(typeof(__traits(getMember, T, m)) == JSONValue!S, S));
+                  enum extrasMember = m;
+              }
```

Comment:
Needs `else` for everything down below.

If a member is the extras member, it's not supposed to be directly usable.

For example:

```d
struct S
{
    int a, b, c;
    @extras JsonValue!string extras;
}
```

Should when parsed with the following:

```json
{ "a" : 1, "b": 2, "c": 3, "d": 4, "extras": "foo"}
```

Should turn into something like:
```d
S(a:1, b:2, c:3, extras: JSONValue!string(["d": 4, "extras": "foo"]);
```

With the way you have it, the `extras` member will be overwritten by the `"foo"` string value.